net/net.go

```diff
@@ -557,14 +559,17 @@ func (n *net) AddReplicator(ctx context.Context, id thread.ID, paddr ma.Multiadd
 	// Send all logs to the new replicator
 	for _, l := range info.Logs {
 		if err = n.server.pushLog(ctx, info.ID, l, pid, info.Key.Service(), nil); err != nil {
-			if err := n.store.SetAddrs(info.ID, ownlg.ID, ownlg.Addrs, pstore.PermanentAddrTTL); err != nil {
-				log.Errorf("error rolling back log address change: %s", err)
+			// Rollback all potentially failed log(s)
+			for _, lg := range logs {
+				if err := n.store.SetAddrs(info.ID, lg.ID, lg.Addrs, pstore.PermanentAddrTTL); err != nil {
+					log.Errorf("error rolling back log address change: %s", err)
+				}
 			}
 			return
```

Comment:
Noticed this early out here. Do we want to fail and roll-back if _any_ of the logs fail to push?

Reply:
If one of the _direct_ logs makes it to the replicator, _technically it does have a new address_, so maybe that shouldn't be rolled back, but still return the error from `pushLog` (as it is now). In other words, I'd vote for rolling back just the failed pushes of _direct_ logs. Sound right?